js/app.js

```diff
@@ -118,15 +118,30 @@ TextApp.prototype.initEditor_ = function() {
   // set up all dependent objects
   this.dialogController_ = new DialogController($('#dialog-container'),
                                                 this.editor_);
-  this.tabs_ = new Tabs(this.editor_, this.dialogController_, this.settings_);
+  if (!this.tabs_) {
+    this.tabs_ = new Tabs(this.editor_, this.dialogController_, this.settings_);
+    this.menuController_ = new MenuController(this.tabs_);
+  } else {
+    // if tabs already exists, just replace the editor so we preserve state
+    this.tabs_.updateEditor(this.editor_);
+  }
 
-  this.menuController_ = new MenuController(this.tabs_);
-  this.searchController_ = new SearchController(this.editor_.getSearch());
-  this.settingsController_ = new SettingsController(this.settings_);
-  this.windowController_ = new WindowController(
+  if (!this.windowController_) {
+    this.windowController_ = new WindowController(
       this.editor_, this.settings_, this.analytics_, this.tabs_);
-  this.hotkeysController_ = new HotkeysController( this.windowController_,
+  } else {
+    this.windowController_.updateEditor(this.editor_);
+  }
+
+  if (!this.hotkeysController_) {
+    this.hotkeysController_ = new HotkeysController( this.windowController_,
       this.tabs_, this.editor_, this.settings_, this.analytics_);
+  } else {
+    this.hotkeysController_.updateEditor(this.editor_);
+  }
+
+
+  this.searchController_ = new SearchController(this.editor_.getSearch());
```

Comment:
Good question! I'll chuck a comment to explain it better. Long story short some objects need to be updated rather then recreated otherwise we lose all of our tabs / open files on the switch to/from a11y mode